File: 02 백준/01 자료구조/Python/Prob18258.py

"""
title  : 18258. 큐 2 (Silver 4)
time   : 1252ms      936ms
memory : 175120KB    253912KB
"""

import sys
from collections import deque

def solution():

    input = sys.stdin.readline
    N = int(input())                # 명령 수

    queue = deque()

    for _ in range(N):

        command = input().split()

        if command[0] == "push":
            queue.append(command[1])
        elif command[0] == "pop":
            print(queue.popleft() if queue else -1)
        elif command[0] == "size":
            print(len(queue))
        elif command[0] == "empty":
            print(0 if queue else 1)
        elif command[0] == "front":
            print(queue[0] if queue else -1)
        elif command[0] == "back":
            print(queue[-1] if queue else -1)
            
solution()

File: 02 백준/01 자료구조/Java/src/Prob18258.java

/**
* title  : 18258. 큐 2 (Silver 4)
* time   : 1220ms
* memory : 407668KB
*/

import java.io.*;
import java.util.*;

public class Prob18258 {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        Deque<Integer> q = new LinkedList<>();

        int N = Integer.parseInt(br.readLine());
        int lastElement = 0;

        for (int i = 0; i < N; i++) {

            String[] comm = (br.readLine()).split(" ");

            if (comm[0].equals("push")) {
                int element = Integer.parseInt(comm[1]);
                q.offer(element);
                lastElement = element;

            } else if (comm[0].equals("pop")) {
                sb.append(q.isEmpty() ? -1 : q.poll()).append("\n");

            } else if (comm[0].equals("size")) {
                sb.append(q.size()).append("\n");
        
            } else if (comm[0].equals("empty")) {
                sb.append(q.isEmpty() ? 1 : 0).append("\n");

            } else if (comm[0].equals("front")) {
                sb.append(q.isEmpty() ? -1 : q.peek()).append("\n");

            } else if (comm[0].equals("back")) {
                sb.append(q.isEmpty() ? -1 : lastElement).append("\n");
            } 
        }
        
        System.out.println(sb);
    }
}